### PR TITLE
refactor(drainer, router): KMS decrypt database password when `kms` feature is enabled

### PR DESCRIPTION
**Pull request by SanchithHegde**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

This PR enables the use of KMS encrypted database passwords for the `router` and `drainer` crates when the `kms` feature is enabled.

### Additional Changes

- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

I prefer being explicit that a variable expects a KMS-encryted value, so I use `kms_encrypted_password` and KMS decrypt it to obtain the password when the `kms` feature is enabled, and use the `password` config value directly whenever the `kms` feature is disabled. Let me know if it would be preferable to use the `password` config value in both scenarios.

When `kms` feature is enabled, the `ROUTER__MASTER_DATABASE__PASSWORD`, `ROUTER__REPLICA_DATABASE__PASSWORD` and `DRAINER__MASTER_DATABASE__PASSWORD` environment variables should now be replaced with `ROUTER__MASTER_DATABASE__KMS_ENCRYPTED_PASSWORD`, `ROUTER__REPLICA_DATABASE__KMS_ENCRYPTED_PASSWORD` and `DRAINER__MASTER_DATABASE__KMS_ENCRYPTED_PASSWORD` instead, while ensuring that their values are replaced with their KMS encrypted equivalents.

KMS encrypted database passwords can be obtained like so:

```shell
$ echo -n 'MyVerySecretDatabasePassword' | base64

# Use the obtained ciphertext as the value for the config variable
$ aws kms encrypt --output json --key-id my-aws-key-id --plaintext <BASE64_ENCODED_DATABASE_PASSWORD>
```

## Motivation and Context

This should add another layer of security for the database passwords.

## How did you test it?

Manually.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: config/config.example.toml
Comment (jarnura) [inline config/config.example.toml:31]: move all `aws_key_id` and `aws_region` to single place under kms itself.

**Review by jarnura — CHANGES_REQUESTED** [review]

File: config/config.example.toml
Comment (SanchithHegde) [reply config/config.example.toml:31]: I've opened #744 for this change. Marking this PR as blocked until it gets merged.

**Comment by SanchithHegde** [comment]
Rebased and force pushed. The drainer database password is also KMS decrypted at startup now.

**Review by ashokkjag — APPROVED** [review]

File: crates/drainer/src/connection.rs
Comment (SanchithHegde) [reply crates/drainer/src/connection.rs:35]: Unfortunately, the `diesel::r2d2::ConnectionManager` stores the connection string.

https://github.com/diesel-rs/diesel/blob/3d68571069e0f28582ae20e2120482203243bd55/diesel/src/r2d2.rs#L121

**Pull request merged**
